src/main/java/org/folio/inventory/consortium/consumers/ConsortiumInstanceSharingHandler.java

```diff
@@ -0,0 +1,277 @@
+package org.folio.inventory.consortium.consumers;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import io.vertx.kafka.client.producer.KafkaHeader;
+import io.vertx.kafka.client.producer.KafkaProducer;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.consortium.model.ConsortiumEnumStatus;
+import org.folio.inventory.consortium.model.ConsortiumEvenType;
+import org.folio.inventory.consortium.model.SharingInstance;
+import org.folio.inventory.dataimport.consumers.DataImportKafkaHandler;
+import org.folio.inventory.dataimport.exceptions.OptimisticLockingException;
+import org.folio.inventory.dataimport.handlers.matching.util.EventHandlingUtil;
+import org.folio.inventory.domain.instances.Instance;
+import org.folio.inventory.domain.instances.InstanceCollection;
+import org.folio.inventory.exceptions.NotFoundException;
+import org.folio.inventory.storage.Storage;
+import org.folio.kafka.AsyncRecordHandler;
+import org.folio.kafka.KafkaConfig;
+import org.folio.kafka.KafkaHeaderUtils;
+import org.folio.kafka.KafkaTopicNameHelper;
+import org.folio.kafka.exception.DuplicateEventException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang.StringUtils.EMPTY;
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+import static org.folio.inventory.consortium.model.ConsortiumEnumStatus.COMPLETE;
+import static org.folio.inventory.consortium.model.ConsortiumEnumStatus.ERROR;
+import static org.folio.inventory.dataimport.util.DataImportConstants.UNIQUE_ID_ERROR_MESSAGE;
+
+public class ConsortiumInstanceSharingHandler implements AsyncRecordHandler<String, String> {
+
+  private static final Logger LOGGER = LogManager.getLogger(DataImportKafkaHandler.class);
+
+  private static final String OKAPI_TOKEN_HEADER = "x-okapi-token";
```

Comment:
I believe that are some constants in RMB that already store these values. It would be good to reuse them.

Reply:
I couldn't find these constants

Reply:
Olamide Aliaks, mod-inventory doesn't use RMB that's why these constants might not be present here

Reply:
I am able to reference it in mod-inventory code, so I assume it is pulled in by some transitive dependency.

Reply:
![image](https://github.com/folio-org/mod-inventory/assets/8954252/f51a2076-cfc3-45e0-8c5f-3d8a37134dd4)